src/Commands/Modules/RenameModule.php

```diff
@@ -0,0 +1,322 @@
+                    $io->success('The module ' . $newModuleName . ' has been uninstalled.');
+                } else {
+                    $io->error('The module ' . $newModuleName . ' couldn\'t be uninstalled.');
+
+                    return 1;
+                }
+            }
+            exec('rm -rf ' . $newFolderPath);
+        }
+
+        if (!$keepOld && $oldModule && $moduleManager->isInstalled($oldModuleName)) {
+            if ($oldModule->uninstall()) {
+                $io->success('The old module ' . $oldModuleName . ' has been uninstalled.');
+            } else {
+                $io->error('The old module ' . $oldModuleName . ' couldn\'t be uninstalled.');
+
+                return 1;
+            }
+        }
+        exec('cp -R ' . $oldFolderPath . '. ' . $newFolderPath);
+        if (!$keepOld) {
+            exec('rm -rf ' . $oldFolderPath);
+        }
+
+        $finder = new Finder();
+        $finder->exclude(['vendor', 'node_modules']);
+        foreach ($finder->in($newFolderPath) as $file) {
+            if ($file->isFile()) {
+                $fileContent = file_get_contents($file->getPathname());
+                file_put_contents($file->getPathname(), strtr($fileContent, $replace_pairs));
+            }
+            foreach ($replace_pairs as $search => $replace) {
+                if (strpos($file->getRelativePathname(), $search) !== false) {
+                    rename($file->getPathname(), $newFolderPath . strtr($file->getRelativePathname(), $replace_pairs));
+                    break;
+                }
+            }
+        }
+
+        // Composer\Factory::getHomeDir() method
+        // needs COMPOSER_HOME environment variable set
+        //putenv('COMPOSER_HOME=' . __DIR__ . '/vendor/bin/composer');
+
+        chdir($newFolderPath);
+        exec('rm -rf vendor');
+        $installCommand = new ArrayInput(['command' => 'update']);
+        $dumpautoloadCommand = new ArrayInput(['command' => 'dumpautoload', '-a']);
+        $application = new Application();
+        $application->setAutoExit(false);
+        $application->run($installCommand, new NullOutput());
+        $application->run($dumpautoloadCommand, new NullOutput());
+        chdir('../..');
+
+        $newModuleName = strtolower($newFullName['prefix'] . $newFullName['name']);
+        $newModule = Module::getInstanceByName($newModuleName);
+        if ($newModule && $newModule->install()) {
+            $io->success('The fresh module ' . $newModuleName . ' has been installed.');
+        } else {
+            $io->error('The module ' . $oldModuleName . ' couldn\'t be installed.');
+        }
+
+        return 0;
+    }
+
+    public function mapModuleName($arg)
+    {
+        if (!$arg) {
+            return false;
+        }
+
+        $fullName = explode('_', $arg);
+        if (count($fullName) > 2) {
+            return false;
+        }
+
+        $prefix = count($fullName) == 2 ? $fullName[0] : '';
```

Comment:
Since I try to respect the latest PrestaShop conventions as much as possible (and I would like to encourage people to do so), I have decided to comply with the requirements of the new translation system:

> Only word characters (A-z, 0-9) are supported.
> Make sure that the technical name of your module does not contain contains underscores (_) or any other unsupported symbol, or else translation may not work.

Should I add an `underscore` option to make add one in the new module name?

Reply:
No you are right :+1:  but in the arguments i see : 

`Module new name with following format : Prefix_ModuleNewNameCamelCased`

I just tested again with : `../../bin/console fop:modules:rename ps_emailsubscription okommail --keep-old`

`Ps_Emailsubscription` is not proposed for the  occurence i think this bug is only with prestashop modules.

![Capture d’écran de 2021-07-11 16-57-39](https://user-images.githubusercontent.com/5064590/125200015-3a26d900-e269-11eb-8b2f-e8053b9164f3.png)

Reply:
Ah no, this is not a bug!
You have to camel case the names, otherwise the command won't know the different "words" of the module. For this one, it's not so important but with Ps_CustomerSignIn for example, it is.
So with ps_emailsubscription, here is the right command (assuming you want okom as prefix and mail as name for the new module) :

`../../bin/console fop:modules:rename Ps_Emailsubscription okom_mail --keep-old`

I get the following result:

![image](https://user-images.githubusercontent.com/48719560/125201856-3bf49a80-e271-11eb-9373-54e2c8b6cb70.png)

The fresh module is installed on my side, it should work if you camel case the module names.

Reply:
Okay now I understand how it works :+1: 

Whyt not for the old module to say use the class name because for me module name is `$this->name = 'ps_emailsubscription';`

It's work good very nice job.

Reply:
Well, I thought about it too but if the old module name has no underscore in its full name, how can we know where the prefix ends? Here are some examples, where the prefix is "Fop" and the name is "ModuleExample" :

- FopModuleExample
- FOPModuleExample
- FoPModuleExample

I could assume that everyone uses the same format in their module prefix, but I prefer to let the user choose their format freely. I added warnings if names are not camel cased (no upper case letter in the full name).